plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -261,9 +261,70 @@ export class EventsProcessor {
             ...(historicalMigration ? { historical_migration: true } : {}),
         }
 
+        const team = await this.teamManager.getTeam(teamId)
+        if (team) {
+            this.extractDynamicMaterializedColumns(rawEvent, properties, team)
+        }
+
         return rawEvent
     }
 
+    private extractDynamicMaterializedColumns(rawEvent: RawKafkaEvent, properties: Properties, team: Team): void {
+        if (!team.materialized_column_slots || team.materialized_column_slots.length === 0) {
+            return
+        }
+
+        for (const slot of team.materialized_column_slots) {
+            // Only process READY and BACKFILL slots (skip ERROR)
+            if (slot.state !== 'READY' && slot.state !== 'BACKFILL') {
+                continue
+            }
+
+            const propertyValue = properties[slot.property_name]
+            if (propertyValue === undefined || propertyValue === null) {
+                continue
+            }
+
+            const columnName = `dmat_${slot.slot_property_type}_${slot.slot_index}` as keyof RawKafkaEvent
+            const convertedValue = this.convertPropertyValue(propertyValue, slot.slot_property_type)
+
+            if (convertedValue !== null) {
+                ;(rawEvent as any)[columnName] = convertedValue
```

Comment:
question: Do we need the cast as any here? Skipping the type checks has caused us numerous issues in the past and they've all been really painful to debug.